### DISCUSSION TITLE
merge two datasets gives unexpected dimensions

### DISCUSSION THREAD

**Discussion by shuai-zhou** [pydata/xarray#6158] [Q&A] [answered]
I have two datasets, one is `ds_prcp` (precipitation data), another is `ds_tmean` (mean temperature data), both have exactly the same dimensions `(longitude: 1405, latitude: 621, t: 12)`

![1](https://user-images.githubusercontent.com/22311821/149021912-e1aba038-2960-4dd3-a29c-5a21916c0b7d.png)

I want to combine the two datasets using the `merge` method: `ds = xr.merge([ds_prcp, ds_tmean])`, but it gives the wrong dimensions:

![2](https://user-images.githubusercontent.com/22311821/149022206-390a8d8a-60d4-4e36-9124-d015e8b5d059.png)

The `dimensions` should be still `(longitude: 1405, latitude: 621, t: 12)`. What is the right way to do this? Thanks.

**Comment by keewis** [comment, marked as answer]
just a quick guess: maybe your coordinates are not exactly equal? In other words, I think this is a floating-point issue.

To check that, try `ds_prcp.longitude.equals(ds_tmean.longitude)`, and you can also try comparing the output of `ds_prcp.longitude.item(0)` with `ds_tmean.longitude.item(0)`.

If you are sure that the coordinates should be exactly equal, you can override the coordinates of one dataset, e.g.:
```python
ds_tmean = ds_tmean.assign_coords(longitude=ds_prcp.longitude, latitude=ds_prcp.latitude)
```

**Reply by dcherian** [reply]
Passing `join="override"` to `merge` will do what keewis suggested.

**Reply by shuai-zhou** [reply]
Thanks to both suggestions. Passing `join="override"` does the work! Thanks.
